addons/account/tests/test_account_move_out_invoice.py

```diff
@@ -2706,3 +2706,9 @@ def _check_invoice_values(invoice):
         invoice_onchange = move_form.save()
 
         _check_invoice_values(invoice_onchange)
+
+    def test_out_invoice_multiple_switch_payment_terms(self):
+        # assertNotUnbalancedEntryWhenSaving
+        with Form(self.invoice) as move_form:
+            move_form.partner_id = self.partner_b # Switch to 30% in advance payment terms
+            move_form.partner_id = self.partner_a # Back to immediate payment term
```

Comment:
Does this test anything? :thinking: 
Also, why not setting the payment term directly? It'd be more robust if the tests get refactored in the future (by you, for example :p).

Reply:
Good point :)

Unfortunately, `assertNotRaises` does not exists :(

Reply:
We could add it if it is clearer, but it would just be an empty context manager...